modules/dxp/apps/commerce/commerce-machine-learning-test/src/testIntegration/java/com/liferay/commerce/machine/learning/recommendation/test/ProductContentCommerceMLRecommendationManagerTest.java

```diff
@@ -0,0 +1,192 @@
+ *
+ *
+ *
+ */
+
+package com.liferay.commerce.machine.learning.recommendation.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.commerce.machine.learning.recommendation.ProductContentCommerceMLRecommendation;
+import com.liferay.commerce.machine.learning.recommendation.ProductContentCommerceMLRecommendationManager;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.CompanyTestUtil;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.search.test.util.IdempotentRetryAssert;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Riccardo Ferrari
+ */
+@RunWith(Arquillian.class)
+public class ProductContentCommerceMLRecommendationManagerTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Before
+	public void setUp() throws Exception {
+		_company = CompanyTestUtil.addCompany();
+
+		_productContentCommerceMLRecommendations = _populateEntries(4, 11);
+	}
+
+	@Test
+	public void testGetProductContentCommerceMLRecommendations()
+		throws Exception {
+
+		Stream<ProductContentCommerceMLRecommendation>
+			productContentCommerceMLRecommendationStream =
+				_productContentCommerceMLRecommendations.stream();
+
+		List<ProductContentCommerceMLRecommendation>
+			expectedProductContentCommerceMLRecommendationList =
+				productContentCommerceMLRecommendationStream.filter(
+					recommendation -> recommendation.getEntryClassPK() == 2
+				).sorted(
+					Comparator.comparingInt(
+						ProductContentCommerceMLRecommendation::getRank)
+				).collect(
+					Collectors.toList()
+				);
+
+		IdempotentRetryAssert.retryAssert(
+			3, TimeUnit.SECONDS,
+			() -> {
+				_assetResultEquals(
+					2, expectedProductContentCommerceMLRecommendationList);
+
+				return null;
+			});
+	}
+
+	private void _assetResultEquals(
+			long entryClassPK,
+			List<ProductContentCommerceMLRecommendation>
+				expectedProductContentCommerceMLRecommendationList)
+		throws Exception {
+
+		List<ProductContentCommerceMLRecommendation>
+			productContentCommerceMLRecommendations =
+				_productContentCommerceMLRecommendationManager.
+					getProductContentCommerceMLRecommendations(
+						_company.getCompanyId(), entryClassPK);
+
+		int expectedRecommendationListSize = Math.min(
```

Comment:
I think we can name it this to follow the changes above, `expectedRecomendationsSize`

Reply:
Done, thanks